script.js

```diff
@@ -295,27 +325,49 @@ async function next() {
     'page_path': '/next/' + state.lang.speechTag,
   });
   try {
-    console.info('Finding article.');
-    let article = await getArticleForLocation();
-    if (!article) {
-      console.info('Did not find location article. Trying nearby.');
-      article = await getNearbyArticle();
-    }
     if (!article) {
-      console.info('Did not find article');
-      setTimeout(next, pollSeconds * 1000);
-      return;
+      try {
+        console.info('Finding article.');
+        article = await getArticleForLocation()
+
+      } catch(e) {
+        try {
+          console.info('Did not find location article. Trying nearby.');
+          article = await getNearbyArticle();
+
+        } catch (e) {
+          console.info('Did not find article');
+          setTimeout(next, pollSeconds * 1000);
+        }
+      }
     }
+
     console.info('Retrieved article');
     state.loading = false;
     render();
     await talkAboutLocation(article);
+
   } catch (e) {
     console.error('Error :' + e, e);
     setTimeout(next, pollSeconds * 1000);
     return;
   }
-  next();
+  next(state.interjectArticle);
+}
```

Comment:
allow the `next` recursive loop to be interjected by an article from another source, e.g. user selected saved article